### PR TITLE
Update steps for 'Getting the source code' and link 'Git Bootcamp'

### PR DESCRIPTION
**Pull request by abdnafees**
This PR contains the following changes mentioned in [issue number #964](https://github.com/python/devguide/issues/964):

- list all upstream-related commands in a single step and code block 
- have the same commands in the boot camp, but with more in-depth discussions (e.g. why they are necessary)
- from here, link to the boot camp so that people that want more details can find them there
- add a couple of comments about the duplicated commands as a reminder for people updating either page

### CODE REVIEW

**Review by ezio-melotti — CHANGES_REQUESTED** [review]
Thanks for the PR!

There are two `.DS_Store` files that got included accidentally and should be removed.
I usually use `git add -p` while adding changes, in order to both verify exactly what gets added and to review it while I do (and sometimes also to break changes into multiple commits).

**Comment by abdnafees** [comment]
Sorry about that let me remove them.

**Comment by abdnafees** [comment]
Would it be useful to add `.DS_Store` to `.gitignore`?

**Comment by abdnafees** [comment]
I hope it worked, I can't see the files now. Please confirm.

File: .gitignore [hunk 1]
```diff
@@ -89,4 +89,4 @@ venv/
 .spyderproject
 
 # Rope project settings
-.ropeproject
+.ropeproject
```
Comment (Mariatta) [inline .gitignore:92]: We need a newline at the end of the file.
```suggestion
.ropeproject

```

Comment (abdnafees) [reply .gitignore:92]: Alright, let me add it.

File: getting-started/setup-building.rst [hunk 2]
```diff
@@ -76,21 +76,16 @@ You will only need to execute these steps once per machine:
 
    (You can use both SSH-based or HTTPS-based URLs.)
 
-6. Configure an ``upstream`` remote::
+.. Step 6 and 7 are are duplicated in bootcamp as well. Please update these steps in both places.
+
+6. Add an ``upstream`` remote, then configure ``git`` to pull ``main`` from ``upstream`` and always push to ``origin``::
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:81]:
```suggestion
6. Add an ``upstream`` remote, then configure ``git``
   to pull ``main`` from ``upstream`` and always push to ``origin``::
```

File: .gitignore [hunk 3]
```diff
@@ -89,4 +89,4 @@ venv/
 .spyderproject
 
 # Rope project settings
-.ropeproject
\ No newline at end of file
+.ropeproject
```
Comment (abdnafees) [inline .gitignore:92]: Why doesn't it show the new line I added? 
Screenshot 2022-11-16 at 01 56 22

File: getting-started/setup-building.rst [hunk 4]
```diff
@@ -76,21 +76,16 @@ You will only need to execute these steps once per machine:
 
    (You can use both SSH-based or HTTPS-based URLs.)
 
-6. Configure an ``upstream`` remote::
+.. Step 6 and 7 are are duplicated in bootcamp as well. Please update these steps in both places.
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:79]:
```suggestion
.. Step 6 and 7 are are duplicated in bootcamp as well.
   Please update these steps in both places.
```

File: getting-started/setup-building.rst [hunk 5]
```diff
@@ -100,6 +95,8 @@ You will only need to execute these steps once per machine:
       $ git config branch.main.remote
       upstream
 
+For more information about these commands see :ref:`Git Bootcamp and Cheat Sheat <git-boot-camp>`.
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:98]:
```suggestion
For more information about these commands see
:ref:`Git Bootcamp and Cheat Sheat <git-boot-camp>`.
```

File: getting-started/git-boot-camp.rst [hunk 6]
```diff
@@ -61,6 +61,8 @@ You can also use SSH-based or HTTPS-based URLs.
 Configure the Remotes
 ---------------------
 
+.. These steps are duplicated in setup-building in step 6 and 7. Please update these there as well.
```
Comment (ezio-melotti) [inline getting-started/git-boot-camp.rst:64]:
```suggestion
.. These steps are duplicated in setup-building in step 6 and 7.
   Please update these there as well.
```

File: .gitignore (hunk 3 above)
Comment (abdnafees) [reply .gitignore]: Got it, thank you for the help.

**Review by ezio-melotti — APPROVED** [review]

**Comment by ezio-melotti** [comment]
Done, thanks for the PR!

**Pull request merged**